crates/parser/src/ast.rs

```diff
@@ -514,6 +520,16 @@ impl<'a> Type<'a> {
                 Ok(Type::Expected(Expected { ok, err }))
             }
 
+            // stream<T, Z>
+            Some((_span, Token::Stream)) => {
+                tokens.expect(Token::LessThan)?;
+                let element = Box::new(Type::parse(tokens)?);
+                tokens.expect(Token::Comma)?;
+                let end = Box::new(Type::parse(tokens)?);
+                tokens.expect(Token::GreaterThan)?;
+                Ok(Type::Stream(Stream { element, end }))
+            }
```

Comment:
Should syntax also be added for `stream<T>` where the `Z` is implicity `unit`?

Reply:
My inclination is to leave it explicit for now, and add implicit syntax later if needed.